### PR TITLE
Add failing python-tuf client test

### LINKED ISSUE #242: conformance: Add python/rust TUF client presubmits to published repository
**Description**

### PR DESCRIPTION
**Pull request by joshuagl**
#### Summary

Adds a simple python-tuf client in tests/client-tests/python-tuf.py, the client currently fails due to #329 which will be resolved with the v5 root.

Fixes #242 

#### Release Note

NONE

### CODE REVIEW

File: tests/client-tests/python-tuf.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    shutil.copyfile(
+        "repository/repository/4.root.json",
+        f"{METADATA_DIR}/root.json")
+
+    updater = Updater(
+        metadata_dir=METADATA_DIR,
+        metadata_base_url=f"{REPO_URL}/metadata/",
+        target_base_url=f"{REPO_URL}/targets/",
+        target_dir=tmpdirname)
+    updater.refresh()
```
Comment (asraa) [inline tests/client-tests/python-tuf.py:42]: TBD (not blocking this PR) on all the compatibility tests: we should consider verifying some properties of the retrieved metadata; but in some way that will pick up on changes to the configuration.

Alternatively, we can co-opt [this](https://github.com/sigstore/root-signing/blob/main/.github/workflows/cosign-test.yml) workflow to be sigstore-client-test.yml and test verification on a stable artifact on all clients (cosign, sigstore-rs, eventually sigstore-python). I will take an AI to look into sigstore-rs and see if the mirror is configurable.

File: .github/workflows/tuf_client_tests.yml [hunk 2]
```diff
@@ -46,3 +46,7 @@ jobs:
             --root repository/repository/2.root.json \
             -t http://localhost:8001/targets \
             -m http://localhost:8001
+      # Test with python-tuf ngclient
```
Comment (asraa) [inline .github/workflows/tuf_client_tests.yml:49]: perfect! Can we merge this and ignore failure? https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepscontinue-on-error

With a follow-up issue marked with milestone v5 to clean-up so we don't forget :)

**Review by asraa — COMMENTED** [review]
thank you so much!

File: tests/client-tests/python-tuf.py (hunk 1 above)
Comment (joshuagl) [reply tests/client-tests/python-tuf.py:42]: Agree we should do more verification of the retrieved metadata. Having a generic sigstore-client-test, as you've suggested, seems like the right approach.

Comment (asraa) [reply tests/client-tests/python-tuf.py:42]: Filed https://github.com/sigstore/root-signing/issues/345 for a v5 target

**Review by asraa — APPROVED** [review]

File: .github/workflows/tuf_client_tests.yml (hunk 2 above)
Comment (joshuagl) [reply .github/workflows/tuf_client_tests.yml:49]: Added in [c3045d6](https://github.com/sigstore/root-signing/pull/343/commits/c3045d6815d837e9d11edeab4d91809bb5feaa46) and files https://github.com/sigstore/root-signing/issues/346 to make the updates

**Pull request merged**